core/base/document.php

```diff
@@ -1801,6 +1859,53 @@ protected function print_elements( $elements_data ) {
 		}
 	}
 
+	public function set_document_cache( $value ) {
```

Comment:
This file is too big. Can we extract cache manipulation to a separate class or trait?

Reply:
maybe it should be within a cache module, I'm not sure

Reply:
I thought about extracting it to the caching module. But we have a lots of codes connected to this class, so I just put it here for now.